### ISSUE TITLE
Missing ClassVar assignment in subclass not flagged

### ISSUE THREAD

**Issue by matejcik** [microsoft/pyright#3652]
**Describe the bug**
The following code typechecks, but is obviously invalid at run-time:
```python
from typing import ClassVar

class Abstr:
    TITLE: ClassVar[str]

    def print_title(self) -> None:
        print(self.TITLE)

class Concr(Abstr):
    TITLE = "Concrete"

class MissingTitle(Abstr):
    pass

a = Concr()
a.print_title()

b = MissingTitle()
b.print_title()  # AttributeError: 'MissingTitle' object has no attribute 'TITLE'
```

**Expected behavior**
Pyright should either flag the `ClassVar` definition site, or instantiation of `MissingTitle`, for having an invalid value for the declared ClassVar

**Comment by erictraut** [comment]
Pyright will catch this error if you define the base class as a `Protocol`. Mypy also reports the error.

I think that's the correct answer here. Make the following small change to your code, and this error will be caught.

```python
class Abstr(Protocol): ...
```
